libs/browser-wallet/src/frontend/hooks/suggest-mnemonic/index.ts

```diff
@@ -2,6 +2,57 @@ import { useCallback, useEffect, useState } from 'react';
 
 import { useJsonRpcClient } from '@/contexts/json-rpc/json-rpc-context';
 import { RpcMethods } from '@/lib/client-rpc-methods';
+import { useSignTypedData } from 'wagmi';
+
+/**
+ * Derives a mnemonic from the user's connected Ethereum wallet
+ */
+export const useDeriveMnemonic = (chainId: number) => {
+  const { signTypedDataAsync } = useSignTypedData();
+  const { request } = useJsonRpcClient();
+  const [error, setError] = useState<string | null>(null);
```

Comment:
I don't think you need these error and loading states. wagmi hooks leverage react-query, so if you look at the returned data from the hook you will see lots of helpful state

Reply:
Discussed on slack, we want to move everything to using react query where possible but will do as a follow up task